### DISCUSSION TITLE
Is there a way to build a pipeline that run test scripts after deployment via Sync?

### DISCUSSION THREAD

**Discussion by wjstk16** [argoproj/argo-cd#6782] [Q&A]
Hi all,

I am building a CI/CD system by integrating Jenkins and argocd.
I'm focusing on the CD system not CI, and if possible, I'd like to configure only argocd except Jenkins.

Here, I have some problems
1. After deploying K8S manifest through Argocd, I need to run test script. This is insufficient for readness/liveness probe in K8S.
SO, is there a way to build a pipeline that run test scripts after deployment via Sync? (Only Argocd environment)

2. Is there a way to configure the Jenkins pipeline to perform sync of Argocd?
I think I can include Argocd as part of the Jenkins pipeline if this is possible. This allows me to configure the test stage after the Argocd deployment.

Best Regards,
Won.

**Comment by jgwest** [comment]
1. You would likely need to create an (e.g. bash) script that watched the Application's `.status.` sync and health fields (depending on which you were looking for), and then once it was healthy it would proceed to run the tests in the script.
2. If the Jenkins instance is able to connect to the Argo CD instance (eg they are both on the same network, and the necessary ingresses are exposed) then you should be able to call the Argo CD CLI: `argocd login` then `argocd app sync (app name)`
